raft/src/test/java/org/apache/kafka/raft/EndpointsTest.java

```diff
@@ -0,0 +1,489 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.raft;
+
+import org.apache.kafka.common.message.BeginQuorumEpochRequestData;
+import org.apache.kafka.common.message.BeginQuorumEpochResponseData;
+import org.apache.kafka.common.message.EndQuorumEpochRequestData;
+import org.apache.kafka.common.message.EndQuorumEpochResponseData;
+import org.apache.kafka.common.message.FetchResponseData;
+import org.apache.kafka.common.message.FetchSnapshotResponseData;
+import org.apache.kafka.common.message.VoteResponseData;
+import org.apache.kafka.common.message.VotersRecord;
+import org.apache.kafka.common.network.ListenerName;
+
+import org.junit.jupiter.api.Test;
+
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+final class EndpointsTest {
+    @Test
+    void testAddressWithValidEndpoint() {
+        Map<ListenerName, InetSocketAddress> endpointMap = new HashMap<>();
```

Comment:
BTW we have `Utils.mkMap` that can be used to define this in one statement instead of two 
```
Utils.mkMap` that can be used to define like 
`Utils.mkMap(Utils.mkEntry(ListenerName.normalised("listener"),InetSocketAddress.createUnresolved("localhost", 9092)));
```

or use simple java 

```
new HashMap<>() {{
put(ListenerName.normalised("listener"), InetSocketAddress.createUnresolved("localhost", 9092));
```

Reply:
Also `ListenerName.normalised("listener")` and  `InetSocketAddress.createUnresolved("localhost", 9092)` seem to be duplicated in every test maybe we can define them on top of the test.

Reply:
updated